### ISSUE TITLE
Builds are failing because of nock/httpbin

### ISSUE THREAD

**Issue by Fishbowler** [IcedFrisby/IcedFrisby#125]
Seeing ~all builds erroring with messages about nock, and some spurious response codes or 599s from httpbin. See [Travis](https://travis-ci.org/IcedFrisby/IcedFrisby/builds/401516891) - not all builds fail the same tests in the same node versions, even when just rerun with no code changes.

I span up [a branch that uses httpbin in a docker container](https://github.com/IcedFrisby/IcedFrisby/tree/move_httpbin_to_docker), using the same version of the code that's live, and it's looking fine & healthy.

This looks like it could be a nock bug, a nock race condition, or a nock misuse by us. The second error here doesn't use nock, so looking likely that we've got a nock problem somewhere within our tests. I tried [updating nock](https://travis-ci.org/IcedFrisby/IcedFrisby/builds/402713635), but it hasn't helped.

```
  1) test with httpbin for valid basic auth
       
	[ GET http://httpbin.org/basic-auth/frisby/passwd ]:
      AssertionError: expected 599 to equal 200
      + expected - actual
      -599
      +200
      
      at _expect (lib/icedfrisby.js:439:45)
      at Frisby._invokeExpects (lib/icedfrisby.js:1266:26)
      at start (lib/icedfrisby.js:1249:12)
      at Request.runCallback [as _callback] (lib/icedfrisby.js:1136:16)
      at self.callback (node_modules/request/request.js:186:22)
      at Request.onRequestError (node_modules/request/request.js:878:8)
      at ErroringClientRequest.<anonymous> (node_modules/nock/lib/intercept.js:231:10)
      at process._tickCallback (internal/process/next_tick.js:61:11)
  2) should send all headers when you bootstrap them with parameters
       
	[ GET /headers ]:
     Error: Error parsing JSON string: Unexpected token I in JSON at position 1
	Given: [IcedFrisby] Destination URL may be down or URL is invalid, NetConnectNotAllowedError: Nock: Not allow net connect for "httpbin.org:443/headers"
      at _jsonParse (lib/icedfrisby.js:61:11)
      at _expect (lib/icedfrisby.js:610:19)
      at Frisby._invokeExpects (lib/icedfrisby.js:1266:26)
      at start (lib/icedfrisby.js:1249:12)
      at Request.runCallback [as _callback] (lib/icedfrisby.js:1136:16)
      at self.callback (node_modules/request/request.js:186:22)
      at Request.onRequestError (node_modules/request/request.js:878:8)
      at ErroringClientRequest.<anonymous> (node_modules/nock/lib/intercept.js:231:10)
      at process._tickCallback (internal/process/next_tick.js:61:11)
```

We could:
* Investigate what's broken
* Switch to using Docker permanently

Gut feel is to do both, but would appreciate some input.

**Comment by paulmelnikow** [comment]
Hey @Fishbowler I forgot all about this. I think you'd asked me to dig into it. Would that still be helpful? Are you still interested in pushing this forward if I have some time to contribute and review?

**Comment by Fishbowler** [comment]
According to the notes on #124, I've fixed this, but it was a while ago, and would still need merging.

**Comment by paulmelnikow** [comment]
It looks like I started a review on #124. Let me finish that up now!

**Comment by Fishbowler** [comment]
Fixed by #124.
